snippets/ch03/reader.py

```diff
@@ -148,7 +148,7 @@ def describe(self, fileids=None, categories=None):
         tokens  = nltk.FreqDist()
 
         # Perform single pass over paragraphs, tokenize and count
-        for para in self.paras(fileids, categories):
+        for para in self.tokenize(fileids, categories):
```

Comment:
@danielw2904 ah, it looks like in translating the examples from the book to the chapter snippet, we combined two different examples into one. I'll update the snippet. In the meantime, the following snippet should work correctly; tokenizing at each step:

```python
for para in self.paras(fileids, categories):
    counts['paras'] += 1 
   
    for sent in sent_tokenize(para):
        counts['sents'] += 1 
    
        for word in wordpunct_tokenize(sent):
            counts['words'] += 1 
            tokens[word] += 1 
```